src/xwayland.c

```diff
@@ -349,8 +351,7 @@ handle_destroy(struct wl_listener *listener, void *data)
 	struct view *view = wl_container_of(listener, view, destroy);
 	struct xwayland_view *xwayland_view = xwayland_view_from_view(view);
 	assert(xwayland_view->xwayland_surface->data == view);
-
-	set_surface(view, NULL);
+	assert(!view->surface); /* dissociate is expected first */
```

Comment:
Hmm, this might not hold if called from set_override_redirect. Need to think about it more.